macros/src/codegen.rs

```diff
@@ -224,105 +252,107 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
         let state_data = match sm.state_data.data_types.get(state) {
             Some(st @ Type::Reference(_)) => quote! { state_data: #st, },
             Some(st) => quote! { state_data: &#st, },
             None => quote! {},
         };
 
-        value.iter().for_each(|(event, value)| {
-            // get input state lifetimes
-            let in_state_lifetimes = sm.state_data.lifetimes.get(&value.in_state.to_string()).cloned().unwrap_or_default();
-
-            // get output state lifetimes
-            let out_state_lifetimes = sm.state_data.lifetimes.get(&value.out_state.to_string()).cloned().unwrap_or_default();
-
-            // get event lifetimes
-            let event_lifetimes = sm.event_data.lifetimes.get(event).cloned().unwrap_or_default();
-
-            // combine all lifetimes
-            let mut all_lifetimes = Lifetimes::new();
-            all_lifetimes.extend(&in_state_lifetimes);
-            all_lifetimes.extend(&out_state_lifetimes);
-            all_lifetimes.extend(&event_lifetimes);
-
-            // Create the guard traits for user implementation
-            if let Some(AsyncIdent {ident: guard, is_async}) = &value.guard {
-                let event_data = match sm.event_data.data_types.get(event) {
-                    Some(et @ Type::Reference(_)) => quote! { event_data: #et },
-                    Some(et) => quote! { event_data: &#et },
-                    None => quote! {},
-                };
-
-                let guard_error = if sm.custom_guard_error {
-                    quote! { Self::GuardError }
-                } else {
-                    quote! { () }
-                };
-
-                // Only add the guard if it hasn't been added before
-                if !guard_set.iter().any(|g| g == guard) {
-                    guard_set.push(guard.clone());
+        event_mappings.iter().for_each(|(event, event_mapping)| {
+            event_mapping.transitions.iter().for_each(|transition|{
+                // get input state lifetimes
+                let in_state_lifetimes = sm.state_data.lifetimes.get(&event_mapping.in_state.to_string()).cloned().unwrap_or_default();
+
+                // get output state lifetimes
+                let out_state_lifetimes = sm.state_data.lifetimes.get(&transition.out_state.to_string()).cloned().unwrap_or_default();
+
+                // get event lifetimes
+                let event_lifetimes = sm.event_data.lifetimes.get(event).cloned().unwrap_or_default();
+
+                // combine all lifetimes
+                let mut all_lifetimes = Lifetimes::new();
+                all_lifetimes.extend(&in_state_lifetimes);
+                all_lifetimes.extend(&out_state_lifetimes);
+                all_lifetimes.extend(&event_lifetimes);
+
+                // Create the guard traits for user implementation
+                if let Some(guard_expression) = &transition.guard {
+                    let _ = visit_guards(guard_expression,|guard| {
```

Comment:
Why are we ignoring a result?

Reply:
visit_guards()  -> Result<(), parse::Error> is a helper function to facilitate a traverse of a guard expression.
It does not return any error itself though. If an error produced by the visitor closure, then it is propagated 
by visit_guards(). In this instance, however the closure does not  return any error, so there is nothing to check on exit. Expression `let _ =` is used to disable the warning.